fault_proof/src/lib.rs

```diff
@@ -411,6 +422,116 @@ where
+            // If the last index is reached and still haven't found a defensible game, return `None`.
+            if game_index > latest_game_index {
+                tracing::info!("No defensible games found after checking all games");
+                return Ok(None);
+            }
+
+            game_address = self.fetch_game_address_by_index(game_index).await?;
+            let game = OPSuccinctFaultDisputeGame::new(game_address, self.provider());
+
+            let claim_data = game.claimData().call().await?.claimData_;
+            if claim_data.status != ProposalStatus::Challenged {
+                tracing::debug!(
+                    "Game {:?} at index {:?} is not challenged, not attempting to defend",
+                    game_address,
+                    game_index
+                );
+                game_index += U256::from(1);
+                continue;
+            }
+
+            // Check if the game has already been proven.
+            if claim_data.status == ProposalStatus::ChallengedAndValidProofProvided {
+                tracing::info!(
+                    "Game {:?} at index {:?} has already been proven, not attempting to defend",
+                    game_address,
+                    game_index
+                );
+                game_index += U256::from(1);
+                continue;
+            }
+
+            // Check if the game is still in the proof submission window.
+            let current_timestamp = l2_provider
+                .get_l2_block_by_number(BlockNumberOrTag::Latest)
+                .await?
+                .header
+                .timestamp;
+            let deadline = U256::from(claim_data.deadline).to::<u64>();
+            if deadline < current_timestamp {
+                tracing::info!(
+                    "Game {:?} at index {:?}, deadline {:?} has passed, current timestamp {:?}, not attempting to defend",
+                    game_address,
+                    game_index,
+                    deadline,
+                    current_timestamp
+                );
+                game_index += U256::from(1);
+                continue;
+            }
+
+            // Check if the game has a valid claim.
+            block_number = game.l2BlockNumber().call().await?.l2BlockNumber_;
+            tracing::info!(
+                "Checking if game {:?} at index {:?} for block {:?} is valid",
+                game_address,
+                game_index,
+                block_number
+            );
+            let game_claim = game.rootClaim().call().await?.rootClaim_;
+
+            let output_root = l2_provider
+                .compute_output_root_at_block(block_number)
+                .await?;
+
+            if output_root == game_claim {
+                tracing::info!(
+                    "Output root {:?} at block {:?} is same as game claim {:?}",
+                    output_root,
+                    block_number,
+                    game_claim
+                );
```

Comment:
Break this out into a separate function?

Reply:
The code snippet you mentioned is also duplicated in `get_oldest_challengable_game_address`, so I created a new helper function `get_oldest_game_address` for refac.